protocols/kad/tests/client_mode.rs

```diff
@@ -0,0 +1,84 @@
+use libp2p_identify as identify;
+use libp2p_identity as identity;
+use libp2p_kad::store::MemoryStore;
+use libp2p_kad::{Kademlia, KademliaConfig, KademliaEvent, Mode};
+use libp2p_swarm::Swarm;
+use libp2p_swarm_test::SwarmExt;
+
+#[async_std::test]
+async fn connection_to_node_in_client_mode_does_not_update_routing_table() {
+    let mut client = Swarm::new_ephemeral(MyBehaviour::client);
+    let mut server = Swarm::new_ephemeral(MyBehaviour::server);
+
+    server.listen().await;
+    client.connect(&mut server).await;
+
+    let server_peer_id = *server.local_peer_id();
+
+    match libp2p_swarm_test::drive(&mut client, &mut server).await {
+        (
+            [MyBehaviourEvent::Identify(_), MyBehaviourEvent::Identify(_), MyBehaviourEvent::Kad(KademliaEvent::RoutingUpdated { peer, .. })],
+            [MyBehaviourEvent::Identify(_), MyBehaviourEvent::Identify(_)],
```

Comment:
Does this test actually test that the server never adds the client to its routing table? It might be emitting a `KademliaEvent::RoutingUpdated` right after the two `Identify` events, right?

Off the top of my head, I don't know how to fix this though.

Reply:
I think I ran the test without the implementation and it failed. Will try again though.